routes/projects/update.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * PUT /projects/:username/:projectName
+ * Creates a new project for the given user. A project object is sent in
+ * the form:
+ * {
+ *   name: <string> required,
+ *   username: <string> required,
+ *   isPublic: <boolean> required,
+ *   publishedAt: <dataTime> optional,
+ *   publishLink: <string> optional,
+ *   files: [ <FileObj1>, <FileObj2>... ] required
```

Comment:
@ashleygwilliams Hmm. Projects will be populated by default, but only on the client side since they won't exist as "project objects" until `POST /projects` is hit. At that point, the files haven't been uploaded to the server yet, so I don't see how there can be reference to them.

Unless we want to create a new "project object" every time a user opens thimble. With a generic name maybe?